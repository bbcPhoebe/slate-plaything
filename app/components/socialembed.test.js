import React from 'react';
import { State } from 'slate';
import { Editor } from 'slate-react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Block, { Container, Input, SocialEmbed } from './SocialEmbed';

const BlockContainer = Block(Container)
BlockContainer.displayName = "BlockContainer"
const schema = {
    nodes: {
        'block-type-under-test': BlockContainer
    }
}

const state = State.fromJSON({
        "document": {
        "nodes": [
            {
            "kind": "block",
            "type": "block-type-under-test",
            "data" : {
                contentUri: '1'
            }
            }
        ]
        }
});

const editor = <Editor schema={schema} state={state}/>

describe('Block', () => {
    it('renders a container', () => {    
        const wrapper = mount(<Editor schema={schema} state={state} onChange={() => {}}/>);
        const block = wrapper.find('BlockContainer')
        block.setProps({editor: null})
        // we think it freaks out becuase of editor in Block, which means it is recursive
        // i.e. editor has block and block has editor
        const json = toJson(block, {mode: 'shallow'})
        console.log('wrapper', json )
        expect(json).toMatchSnapshot()
    })

    it('It passes value as the value prop', () => {

        const wrapper = mount(editor);
        expect(wrapper.find(Container).props().value).toEqual('1')
        expect(wrapper.find(Container).props().value).toEqual('1')
    })
})
