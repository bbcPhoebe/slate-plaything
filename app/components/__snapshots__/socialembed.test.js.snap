// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders a container 1`] = `
<Container
  onChange={[Function]}
  value="1"
>
  <SocialEmbed
    clear={[Function]}
    content=""
    contentUri="1"
    error={false}
    onChange={[Function]}
  >
    <div
      className="igm-social-embed block-container "
    >
      <Input
        className="igm-social-embed__uri igm-input"
        onChange={[Function]}
        onClear={[Function]}
        onClick={[Function]}
        placeholder="For example https://twitter.com/EricHolthaus/status/918197423159341056"
        type="text"
        value="1"
      >
        <div
          className="igm-input"
        >
          <span
            className="igm-input--control"
          >
            <input
              className="igm-social-embed__uri igm-input"
              onChange={[Function]}
              onClick={[Function]}
              placeholder="For example https://twitter.com/EricHolthaus/status/918197423159341056"
              type="text"
              value="1"
            />
          </span>
          <button
            className="igm-input--clear"
            onClick={[Function]}
          >
            Ã—
          </button>
        </div>
      </Input>
      <div
        className="igm-social-embed__content"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
    </div>
  </SocialEmbed>
</Container>
`;
